### PR TITLE
Fix web tests for alma9 selenium 4 installations

### LINKED ISSUE #431: CLI inconsistent pid_file/gvar behaviour when using interactive and non interactive commands.
### Description
The cloudscheduler cli differentiates between tty processes and non-tty processes when creating a pid file [here](https://github.com/hep-gc/cloudscheduler/blob/65323a6a208484ab0412e54fbbeeeb9070f861bb/cli/bin/cloudscheduler#L336) in cli/bin/cloudscheduler. When a pid is associated with a pid_file, its gvar is updated [here](https://github.com/hep-gc/cloudscheduler/blob/65323a6a208484ab0412e54fbbeeeb9070f861bb/cli/bin/cloudscheduler#L365) and if it is not associated with a pid_file, its gvar is updated with [update_pid_defaults()](https://github.com/hep-gc/cloudscheduler/blob/65323a6a208484ab0412e54fbbeeeb9070f861bb/cli/bin/csv2_common.py#L1213) instead. When clouds are added from a non-tty process, the pid file is written to. Upon deletions, the pid file is not updated. This causes references to groups or users in the pid file that may stop existing to cause errors when tty processes attempt to operate after certain deletions. 

This issue can be avoided by exclusively calling cloudscheduler commands using non-interactive processes.

### Steps to reproduce
1. Deploy a csv2 instance to a virtual machine using the playbooks in https://github.com/hep-gc/uvic-heprc-ansible-playbooks
2. Ensure the machine you are testing from has the cloudscheduler cli installed
3. Edit the below script to match your configuration
4. Run the script
```
#!/usr/bin/python3
import subprocess

csv2_server = #string
openstack_server = #string
openstack_password = #string
openstack_user = #string

# command[0] = command
# command[1] = True if piped, false if interactive process
commands = [
    # delete group g0 if present, this delete is run from a noninteractive process which seems to update the pid_file appropriately.
    [['cloudscheduler', 'group', 'delete', '-gn', 'g0', '-Y', '-s', 'unit-test'], True],
    # add group g0 from interactive process
    [['cloudscheduler', 'group', 'add', '-gn', 'g0', '-un', 'tester', '-htcf', f'{csv2_server}', '-s', 'unit-test'], False],
    # add group g1 from interactive process
    [['cloudscheduler', 'group', 'add', '-htcf', f'{csv2_server}', '-gn', 'g1', '-un', 'tester', '-s', 'unit-test'], False],
    # add cloud c1 to group g1 from interactive process, writes to pid_file tester:g1
    [['cloudscheduler', 'cloud', 'add', '-ca', f'{openstack_server}', '-cn', 'c1', '-cpw', f'{openstack_password}', '-cP', 'testing', '-cr', 'Victoria', '-cU', f'{openstack_user}', '-ct', 'openstack', '-g', 'g1', '-s', 'unit-test'], False],
    # delete cloud g1::c1 from noninteractive process, does not update pid_file
    [['cloudscheduler', 'cloud', 'delete', '-cn', 'c1', '-g', 'g1', '-s', 'unit-test', '-Y'], True],
    # delete group g1 from noninteractive process
    [['cloudscheduler', 'group', 'delete', '-gn', 'g1', '-s', 'unit-test', '-Y'], True],
    # attempt to delete g0 from interactive process, pid_file is reads the activeuser:group as tester:g1. g1 was deleted so the removal fails.
    [['cloudscheduler', 'group', 'delete', '-gn', 'g0', '-Y', '-s', 'unit-test'], False]
]

for command in commands:
    if command[1] == True:
        res = subprocess.run(command[0], stdout=subprocess.PIPE)
        if res.stdout: print(res.stdout.decode(), end='')
        if res.stderr: print(res.stderr.decode(), end='')
    else:
        subprocess.run(command[0])

exit(0)
```

### Expected Result
The groups, and clouds are created and deleted. The pid file is updated when clouds are deleted. 

### Actual result
Terminal output:
```
Error: "cloudscheduler group delete" cannot delete "g0", group doesn't exist. //This is expected, in subsequent runs, g0 will exist at the start. 
group "g0" successfully added.
group "g1" successfully added.
cloud "g1::c1" successfully added.
cloud "g1::c1" successfully deleted.
group "g1" successfully deleted.
Error: SV-00055 cannot switch to invalid group "g1".
```
pid file:
```
server: unit-test
server_addresses:
  https://elephant96.heprc.uvic.ca:
    tester: g1
```

### PR DESCRIPTION
**Pull request by travkhl**
Enable web tests to run after setting up a ci server using the playbooks [from branch csv2-ci-alma9](https://github.com/hep-gc/uvic-heprc-ansible-playbooks/tree/csv2-ci-alma9).

Note that due to changes in how the opera driver is referenced in selenium, Opera is no longer working with the setup script.

### CODE REVIEW

File: unit_tests/web_tests/test_web_cloud.py [hunk 1]
```diff
@@ -1186,7 +1184,7 @@ def test_web_cloud_metadata_update_priority_by_typing_string(self):
         self.assertTrue(self.page.metadata_priority_popup_exists())
         wta.assertHasNotAttribute('metadata', metadata_name, 'priority', 'invalid-web-test', group=self.gvar['base_group'], metadata_cloud=cloud_name)
 
-    @unittest.skip("Not working in production (issue 319)")
+    # @unittest.skip("Not working in production (issue 319)")
```
Comment (colsond) [inline unit_tests/web_tests/test_web_cloud.py]: delete me

File: unit_tests/web_tests/test_web_status.py [hunk 2]
```diff
@@ -562,7 +562,7 @@ def test_web_status_plot_hide_line(self):
         self.assertFalse(self.page.plot_has_line())
         self.page.click_close_plot()
     
-
+    @unittest.skip("Not Working Issue-437")
```
Comment (colsond) [inline unit_tests/web_tests/test_web_status.py]: revisit setup and monitor vm situation

File: unit_tests/web_tests/test_web_cloud.py (hunk 1 above)
Comment (travkhl) [reply unit_tests/web_tests/test_web_cloud.py]: removed in [d218695](https://github.com/hep-gc/cloudscheduler/pull/432/commits/d21869592fba01146d475921fdf88924e5428334)

File: unit_tests/web_tests/test_web_status.py (hunk 2 above)
Comment (travkhl) [reply unit_tests/web_tests/test_web_status.py]: Will be resolved by hep-gc/uvic-heprc-ansible-playbooks#14

**Comment by travkhl** [comment]
Tests now all pass with the exception of Opera tests.

#### The following tests remain marked to be skipped 
##### tests involving boot volume expecting a json field
```
 test_web_cloud_update_boot_volume
 test_web_cloud_update_boot_volume_invalid_string
 test_web_cloud_update_boot_volume_empty_keypair
 test_web_cloud_update_boot_volume_invalid_keypair
 test_web_cloud_update_boot_volume_gbs_float
 test_web_cloud_update_boot_volume_gbs_string
 test_web_cloud_update_boot_volume_gbs_per_core_float
 test_web_cloud_update_boot_volume_gbs_per_core_string
 test_web_cloud_update_boot_volume_too_long
```
 
##### tests involving amazon cloud
```
test_web_cloud_update_spot_price
test_web_cloud_update_spot_price_string
```
 
##### tests requiring other types of clouds than openstack
```
test_web_image_add_to_cloud
```
##### tests that are marked "Probably takes too long to be worth running"
```
test_web_cloud_update_ram_by_arrows_too_big
test_web_cloud_update_cores_by_arrows_too_big
test_web_cloud_metadata_add_different_priority_by_arrows_too_big
test_web_default_metadata_add_different_priority_by_arrows_too_big
test_web_default_metadata_update_priority_by_arrow_keys_too_big
test_web_setting_update_status_refresh_interval_by_arrow_keys_too_big

```
##### tests attempting to test unclear boundary values
```
test_web_config_update_condor_poller_batch_commit_size_too_big
test_web_config_update_condor_poller_delete_cycle_interval_too_big
test_web_config_update_condor_poller_retire_interval_too_big
test_web_config_update_condor_poller_sleep_interval_command_too_big
test_web_config_update_condor_poller_sleep_interval_condor_gsi_too_big
test_web_config_update_condor_poller_sleep_interval_job_too_big
test_web_config_update_condor_poller_sleep_interval_machine_too_big
test_web_config_update_condor_poller_sleep_interval_worker_gsi_too_big
```
##### tests that rely on removed elements or features
```
test_web_status_plot_open_ec_two
```
##### tests that are anticipating future features/infra
```
test_web_user_update_cert_cn
test_web_user_update_cert_cn_too_long
```
  
#### The following skipped tests have no reason supplied for being skipped and should be investigated
```
test_web_cloud_update_remove_security_group
test_web_cloud_metadata_add_mismatched_file_type
test_web_cloud_metadata_update_contents_mismatched
test_web_default_metadata_add_mismatched_file_type
test_web_default_metadata_update_contents_mismatched
test_web_group_add_search_bar
test_web_group_update_user_add_search_bar
```
